app/js/components/service_status.ts

```diff
@@ -0,0 +1,156 @@
+import * as Chart from 'chart.js';
+import {ChartTooltipItem} from "chart.js";
+
+interface StatesDictionary {
```

Comment:
Also, during my functional test I came across some issues.

1. Loading times are a bit on the slow side. Might be my test setup or..
2. I have the following error in my browser console: TypeError: "exports" is read-only, do you have that too?
3. One of my services is not displayed with a donut, but the canvas is on the page. I can provide you with a service schema dump if you like.

Reply:
With 'states' I really mean the plural form of 'state' not status. It's the state for a given step in the process.

- I haven't encountered slow loading times myself. Maybe because of the animations it looks a bit slow. If I encounter this later on. I will try to fix it.
- The exports TypeError was because the usage of js for the tests and ts for the code itself. I've refactored the tests to also use ts.
- If you could show me the problem I'll look further into the missing graph.